src/transformers/trainer.py

```diff
@@ -3591,20 +3631,23 @@ def training_step(self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor,
             with amp.scale_loss(loss, self.optimizer) as scaled_loss:
                 scaled_loss.backward()
         else:
+            loss *= self.args.gradient_accumulation_steps
             self.accelerator.backward(loss, **kwargs)
 
         return loss.detach() / self.args.gradient_accumulation_steps
 
-    def compute_loss(self, model, inputs, return_outputs=False):
+    def _compute_loss(self, model, inputs, return_outputs=False, num_items_in_batch=None):
```

Comment:
aren't there a lot of downstream trainers that extend this?

Reply:
for example https://github.com/huggingface/trl/blob/a67f2143c38d6520be8735463ce715ad5c281db8/trl/trainer/cpo_trainer.py#L826

Reply:
I'll reverse this then (store as `_compute_loss` and keep `compute_loss`)

Reply:
thanks, btw, @lewtun we should open a PR to add `num_items_in_batch` or `**kwargs` to `compute_loss` otherwise this will break trl too

Reply:
thanks @winglian !